lib/shopify_transporter/shopify/customer.rb

```diff
@@ -57,12 +57,28 @@ def to_csv
       ).freeze
 
       def top_level_row_values
-        base_hash.merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES)).values
+        default_address.merge(top_level_attributes).values_at(*Customer.columns)
+      end
+
+      def top_level_attributes
+        base_hash
+          .merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES))
+          .delete_if { |_, v| v.nil? }
+      end
+
+      def default_address
+        return {} unless at_least_one_address?
+
+        record_hash['addresses'][0].slice(*ADDRESS_ATTRIBUTES)
+      end
+
+      def at_least_one_address?
+        record_hash['addresses'].present? && record_hash['addresses'][0].present?
       end
 
       def address_row_values
-        return [] unless record_hash['addresses']
-        record_hash['addresses'].map do |address_hash|
+        return [] unless record_hash['addresses'].present? && record_hash['addresses'].length > 1
```

Comment:
I really like `at_least_one_address?` above. How do you feel about moving this to `more_than_one_address?`?

Reply:
Will do!